Testing ATL14 gridded land ice download from earthdata

weiji14 opened the pull request:
Adding unit tests for ATL14 data access via Earthdata, using a region over Svalbard (Iceland is smaller, but bounding box overlaps with Greenland which is big).

Towards resolving #288, but also updating tests behind NSIDC login for the ATL06 dataset by updating from v05 to v06.

weiji14 committed:
Add test for ATL14 gridded land ice download from earthdata

weiji14 committed:
Temporarily allow behind Earthdata tests

JessicaS11 commented:
Looks like the issue with adding this test is now #286.

weiji14 commented:
Ok, but I might as well add some unit tests for ATL14 since I've been playing with it recently. Will see what I can do to increase code coverage.

JessicaS11 commented:
> Ok, but I might as well add some unit tests for ATL14 since I've been playing with it recently. Will see what I can do to increase code coverage.

Sounds great (and wasn't suggesting we not add the tests already here).  Just noticed that now that #288 is resolved with updated certificates at NSIDC, the failing Travis errors were "updated" to #286.

weiji14 committed:
Merge branch 'development' into atl14-data-download

JessicaS11 commented:
See #291

weiji14 committed:
Update spatial extent to Svalbard

Not using Iceland's spatial extent, because it overlaps with the Greenland bounding box and causes the unit test to download >1GB of data.

weiji14 committed:
Merge branch 'development' into atl14-data-download

weiji14 committed:
Set pytest verbose mode for tests_behind_NSIDC_API_login

Easier to debug what specific tests went wrong

weiji14 committed:
Use for loop instead of list comprehension to see what keys are missing

Loop through each key one at a time, so that it's easier to debug when keys are missing.

weiji14 committed:
Update ATL06 options json file to v06

Created using `json.dump(obj=obs, fp=open("ATL06v06_options.json", "w"))`.

weiji14 commented on icepyx/tests/ATL06v06_options.json:
Diff between ATL06v05 and ATL06v06 at https://www.diffchecker.com/uCMMViSG/. Main changes are

```diff
- "fileformats": ["TABULAR_ASCII", "NetCDF4-CF", "Shapefile", "NetCDF-3"]
+ "fileformats": ["TABULAR_ASCII", "NetCDF4-CF", "Shapefile"]
- "formatreproj": ["TABULAR_ASCII", "NetCDF4-CF", "Shapefile", "NetCDF-3"]
+ "formatreproj": ["TABULAR_ASCII", "NetCDF4-CF", "Shapefile"]
...
+ "orbit_info/bounding_polygon_lat1", "orbit_info/bounding_polygon_lon1"
```

weiji14 committed:
[nsidc_tests] Disable tests behind NSIDC login on PR branch

Reverts 2feaf5d71b4b07a040cb8bc2c0bce98605fe5e9f.

weiji14 reviewed (commented):
Cool, should be ready for review! Have updated the `test_behind_NSIDC_API_login` to include the new ATL14 (v002) dataset, and also updated the ATL06 schema to v006 for good measure.

weiji14 committed:
Merge branch 'development' into atl14-data-download

weiji14 committed:
[nsidc_tests] Revert "Remove setting of NSIDC_LOGIN environment variable"

This reverts commit a813fa4b733ae5b16045d8e082aab77bfcd65475.

weiji14 committed:
[nsidc_tests] Remove setting of NSIDC_LOGIN environment variable

Shouldn't be needed since $NSIDC_LOGIN is already set?

weiji14 commented:
Hi @rwegener2, could you give this a review if you have time, since you recently worked on the #435 PR?

rwegener2 commented:
> @rwegener2, could you give this a review if you have time

Absolutely, @weiji14. I've been offline for a day or two while moving apartments, but I'll take a look this afternoon.

rwegener2 reviewed (approved):
This looks good to me @weiji14. Nothing really changed with respect to authentication as far as I can tell, was there something specific you wanted a second set of eyes on? The ATL14/06 specific bits I am less familiar with, but from my experience everything looks solid!

rwegener2 committed:
GitHub action UML generation auto-update

JessicaS11 commented on icepyx/tests/test_behind_NSIDC_API_login.py:
@weiji14 If I'm understanding this right, we've added ATL14 as part of the fixture, but then don't actually run any tests on it beyond creating a query object with those parameters (in which case, shouldn't we just create a query object with it elsewhere instead of in `test_behind_NSIDC_API_login.py`). That or the assertions in lines 50-51 will fail because there's no appropriate json file for the ATL14 custom options...

JessicaS11 committed:
Merge branch 'development' into atl14-data-download
